apis/Google.Storage.V1/Google.Storage.V1/SignedUrlUtils.cs

```diff
@@ -0,0 +1,396 @@
+﻿// Copyright 2015 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Google.Storage.V1
+{
+    /// <summary>
+    /// Class which helps create Signed URLs which can be used to provide limited access to specific buckets and objects
+    /// to anyone in possession of the URL, regardless of whether they have a Google account.
+    /// </summary>
+    /// <remarks>
+    /// See https://cloud.google.com/storage/docs/access-control/signed-urls for more information on Signed URLs.
+    /// </remarks>
+    public static class SignedUrlUtils
```

Comment:
This is much cleaner. I actually wanted to have a Stream parameter for the credentials file as well but didn't want to have 8 overloads instead of 4. Now I can add that option without adding more overloads. I'll change this after merging though.